### PR TITLE
test: add test cases for UI atoms components

### PR DESCRIPTION
**Pull request by hel-axelor**
RM#69731

### CODE REVIEW

File: packages/ui/__tests__/components/atoms/Switch.test.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    const wrapper = shallow(<Switch {...props} />);
+    expect(wrapper.prop('value')).toBe(true);
+  });
+
+  it('should call handleToggle function with value true when switching on', () => {
+    const handleToggle = jest.fn();
+    const props = {
+      isEnabled: false,
+      handleToggle,
+    };
```
Comment (lme-axelor) [inline packages/ui/__tests__/components/atoms/Switch.test.js:44]: Please simplify props creation, it is always the same thing. Define handleToggle function at the beginning and give isEnabled directly to component as it's the only props that is modified

File: packages/ui/__tests__/components/atoms/BlockInteractionScreen.test.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    const wrapper = shallow(
+      <BlockInteractionScreen hideHeader>{children}</BlockInteractionScreen>,
+    );
+
+    const container = wrapper.find('View').at(0);
+
+    const topStyle = container.prop('style')?.top;
+
+    expect(topStyle).toBe(0);
+  });
```
[the remark this answers is not in the document]
Comment (hel-axelor) [reply packages/ui/__tests__/components/atoms/BlockInteractionScreen.test.js:65]: Actually, the root cause of the problem is that I forgot a mock for react native StyleSheet in the mocks setup.

File: packages/ui/__tests__/components/atoms/Icon.test.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      onPress: jest.fn(),
+      disabled: true,
+    };
+
+    const wrapper = shallow(<Icon {...props} />);
+    const iconComponent = wrapper.find('TouchableOpacity');
+
+    expect(iconComponent.prop('disabled')).toBe(true);
+
+    wrapper.simulate('press');
```
[the remark this answers is not in the document]
Comment (hel-axelor) [reply packages/ui/__tests__/components/atoms/Icon.test.js:110]: It should not call onPress when it is disabled we should test this and further having two props disabled and touchable is confusing.

[the remark this answers is not in the document]
Comment (lme-axelor) [reply packages/ui/__tests__/components/atoms/Icon.test.js:100]: I totally agree for the two props which are confusing. I checked `disabled` was added for the StopWatch component and is only used in this component so I'm going to create a ticket to remove it. I will assigned it to you.

For the second point, the issue is not coming from the Icon component. With a quick search on internet, I found that the problem is well known. It's TouchableOpacity mock from RN which is the origin of the issue here. So, as the props disabled of the TouchableOpacity is well set to true, I think our test can stop here and not check if function is called or not which depends on RN and not the Icon component.

**Pull request merged**